velox/exec/tests/PlanNodeToStringTest.cpp

```diff
@@ -501,3 +502,46 @@ TEST_F(PlanNodeToStringTest, tableScan) {
         plan->toString(true, false));
   }
 }
+
+TEST_F(PlanNodeToStringTest, window) {
+  std::vector<exec::FunctionSignaturePtr> signatures{
+      exec::FunctionSignatureBuilder()
+          .argumentType("BIGINT")
+          .returnType("BIGINT")
+          .build(),
+  };
+  exec::registerWindowFunction("window1", std::move(signatures), nullptr);
+
+  PlanBuilder::WindowFrame frame = {
+      core::WindowNode::WindowType::kRange,
+      core::WindowNode::BoundType::kPreceding,
+      "10",
+      core::WindowNode::BoundType::kFollowing,
+      "b"};
+  auto plan =
+      PlanBuilder()
+          .tableScan(ROW({"a", "b", "c"}, {VARCHAR(), BIGINT(), BIGINT()}))
+          .window({"a"}, {"b"}, {"window1(c) AS d"}, {frame})
+          .planNode();
+  ASSERT_EQ(
+      "-- Window[partition by: [a] order by: [b ASC NULLS LAST] "
+      "d := window1(ROW[\"c\"]) RANGE 10 PRECEDING ROW[\"b\"] FOLLOWING] "
+      "-> a:VARCHAR, b:BIGINT, c:BIGINT, d:BIGINT\n",
```

Comment:
should there not be TableScan in this output?

Reply:
The output shows 4 columns `-> a:VARCHAR, b:BIGINT, c:BIGINT, d:BIGINT` but the window function shows `a, b, c` as inputs. What are the window output columns? Is there a pass-through list?

Reply:
The second argument to plan->toString() is a parameter indicating recursive printing which is disabled in this invocation. So the tableScan is skipped.

Yes, all input columns are pass-through for windowNode. And new columns are added after those for the function outputs.